[docs] Add HTML input section to TextField page (#45439)
mui/material-ui: docs/data/material/components/text-fields/text-fields.md

{{"demo": "FormPropsTextFields.js"}}

> ## HTML input

Author: @mapache-salvaje do you think the title is too broad? is `Controlling HTML input` better?

> ## HTML input

Reviewer: How about this?

```suggestion
## Controlling the HTML input
```

Use `slotProps.htmlInput` to pass attributes to the underlying `input` element.

```jsx
<TextField slotProps={{ htmlInput: { 'data-testid': '…' } }} />
```

The rendered HTML input will look like this:

```html
<input
  aria-invalid="false"
  class="MuiInputBase-input MuiOutlinedInput-input"
  type="text"
  data-testid="…"
/>
```

> :::info

Author: ```suggestion
:::warning
```

The `slotProps.htmlInput` is **NOT** the same as the `slotProps.input`.

> The former represent the HTML input whereas the latter refers to `Input`, `FilledInput`, or `OutlinedInput` component based on the specified variant prop.

Reviewer: This is a tricky sentence! I might suggest splitting it into two—here's how I'd do it if I'm understanding correctly:

```suggestion
`slotProps.input` refers to the React `<Input />` component that's rendered based on the specified variant prop.
`slotProps.htmlInput` refers to the HTML `<input>` element rendered within that Input component, regardless of the variant.
```

Author: That's perfect.

{{"demo": "FormPropsTextFields.js"}}

## Controlling the HTML input

> Use `slotProps.htmlInput` to pass attributes to the underlying `input` element.

Reviewer: just to make it a little more explicit that it's an HTML element.

```suggestion
Use `slotProps.htmlInput` to pass attributes to the underlying `<input>` element.
```
